libibumad/umad.h

```diff
@@ -54,6 +54,21 @@ typedef __be16 __attribute__((deprecated)) be16_t;
 typedef __be32 __attribute__((deprecated)) be32_t;
 typedef __be64 __attribute__((deprecated)) be64_t;
 
+typedef union _ib_gid
+{
+	uint8_t	raw[16];
+
+	struct {
+		__be64	prefix;
+		__be64	interface_id;
+	} unicast;
+
+	struct {
+		uint8_t	header[2];
+		uint8_t	raw_group_id[14];
+	} multicast;
+} ib_gid_t;
```

Comment:
Yes something like that.

I'm starting to think that what we should do is call this something other than ib_gid_t.  Deprecate ib_gid_t in all headers and then eventually remove ib_gid_t in favor of the new type.

I'm terrible at names but "rdma_gid_t"?